internal/shim-sev/src/allocator.rs

```diff
@@ -242,7 +242,9 @@ impl EnarxAllocator {
                         let new_size: usize = 2u64.checked_mul(last_size as u64).unwrap() as _;
                         let new_size = new_size.min(self.max_alloc);
                         let num_pages = new_size.checked_div(Page4KiB::size() as _).unwrap();
-                        if let Ok(virt_offset) = HOST_CALL.lock().balloon(num_pages) {
+                        if let Ok(virt_offset) =
+                            HOST_CALL_ALLOC.try_alloc().unwrap().balloon(num_pages)
```

Comment:
Are we worried about a panic here if the result is `None`?

Reply:
I thought about this during my reading of it too. If all N blocks are busy while syscall N+1 tries to find a block, each syscall will panic when it unwraps None. However, this situation is extremely unlikely. It would require a keep with a ridiculous number of threads. In cases like this during early development, a panic may be more useful and alert us to syscall congestion issues better than a retry mechanism.

Maybe some day when the shim can sleep a process context thread, it can wake it when a block is available.

Reply:
Thanks @connorkuehl , that makes sense. If we do want it to panic currently, I'd suggest we should get a specific panic output message for debugging if we can. Possibly a debt issue also to change this later on to something like sleep / retry, at least to record that idea.

Reply:
Ack: opened https://github.com/enarx/enarx-keepldr/issues/253